src/toolkits/coreml_export/neural_net_models_exporter.cpp

```diff
@@ -391,21 +392,35 @@ std::shared_ptr<coreml::MLModelWrapper> export_style_transfer_model(
     const neural_net::model_spec& nn_spec, size_t image_width,
     size_t image_height, bool include_flexible_shape,
     flex_dict user_defined_metadata, std::string content_feature,
-    std::string style_feature) {
+    std::string style_feature, size_t num_styles) {
   CoreML::Specification::Model model;
   model.set_specificationversion(3);
 
   ModelDescription* model_desc = model.mutable_description();
 
   FeatureDescription* model_input = model_desc->add_input();
-  ImageFeatureType* input_feat = set_image_feature(model_input, image_width, image_height, content_feature, "Input image");
+  ImageFeatureType* input_feat = set_image_feature(model_input,
+                                                   image_width,
+                                                   image_height,
+                                                   content_feature,
+                                                   "Input image");
 
   set_array_feature(
       model_desc->add_input(), "index",
-      "Style index array (set index I to 1.0 to enable Ith style)", {1});
+      "Style index array (set index I to 1.0 to enable Ith style)", {num_styles});
+  /*
+   * prefix style with stylized and capitalize the following identifier, this
+   * avoids name clashes with the `content_feature` for exporting to CoreML.
+   */
+  style_feature[0] = toupper(style_feature[0]);
```

Comment:
Nit: could we use `std::toupper` to make it more clear this is from the standard library?

Reply:
Yup!